WordPressAuthenticator/UI/SearchTableViewCell.swift

```diff
@@ -62,16 +73,65 @@ private extension SearchTableViewCell {
 //
 extension SearchTableViewCell: UITextFieldDelegate {
     open func textFieldShouldClear(_ textField: UITextField) -> Bool {
-        delegate?.startSearch(for: "")
+        if !liveSearch {
+            delegate?.startSearch(for: "")
+        }
+
         return true
     }
 
     open func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-        if let searchText = textField.text {
+        if !liveSearch,
+           let searchText = textField.text {
             delegate?.startSearch(for: searchText)
         }
+
+        return false
+    }
+
+    open func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        let sanitizedString: String
+
+        if allowSpaces {
+            sanitizedString = string
+        } else {
+            sanitizedString = string.trimmingCharacters(in: .whitespacesAndNewlines)
+        }
+
+        let hasValidEdits = sanitizedString.count > 0 || range.length > 0
+
+        if hasValidEdits {
+            guard let start = textField.position(from: textField.beginningOfDocument, offset: range.location),
+                  let end = textField.position(from: start, offset: range.length),
+                  let textRange = textField.textRange(from: start, to: end) else {
+
+                // This shouldn't really happen but if it does, let's at least let the edit go through
+                return true
+            }
+
+            textField.replace(textRange, withText: sanitizedString)
+            startLiveSearchAfterEdit()
```

Comment:
Should we check for `liveSearch` before calling `startLiveSearchAfterEdit`? I know that method also checks it internally, but I wonder if we shouldn't assume that.

Reply:
It would probably be cleaner to check outside of it, and document it as a pre-condition for calling the method.

I'll also remove the check inside because if it's properly documented as a precondition it shouldn't be checked twice.

Reply:
I've also generalized the method a bit removing references to "after edits" because it's not really a requirement in any way.